### PR TITLE
add missing IDT2 font csp directives

### LINKED ISSUE #6611: Investigate font CSP errors on some Story pages
**Describe the bug**
When you load some Story pages when you open the console you see many failed attempts to load fonts:
image

https://www.test.bbc.com/gahuza/49116789?renderer_env=live

Multiple examples were found when we tested includes:
https://www.test.bbc.com/gahuza/amakuru-50330144?renderer_env=live
https://www.test.bbc.com/gahuza/amakuru-50332056?renderer_env=live

**To Reproduce**
Steps to reproduce the behaviour:

1. Visit one of the urls listed above
1. Open console
Observe font download errors

No obvious visual problem can be seen on the page.

**Expected behaviour**
No font download errors

**Screenshots**

**Desktop (please complete the following information):**

- OS: Mac
- Browser Firefox and Chrome tested, could be more widespread

**Testing notes**

- [X] This bug fix is expected to need manual testing.

**Additional context**
- Reach out @sareh for more context on the issue. (There was a long thread in Slack).
- Some pages do not seem to exhibit the problem: https://www.test.bbc.com/gahuza/amakuru-36544625?renderer_env=live 

It was found during includes testing so could be related to includes.

### PR DESCRIPTION
**Pull request by rhenshaw56**
Resolves #6611

**Overall change:**
_Adds missing font CSP directives._

**Code changes:**

- _add `news.files.bbci.co.uk` and `news.test.files.bbci.co.uk` to font-src CSP off follow up chat with @Tarqwyn ._
---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [X] This PR requires manual testing

### CODE REVIEW

**Review by karinathomasbbc — APPROVED** [review]

File: src/server/utilities/cspHeader/index.js [hunk 1]
```diff
@@ -361,14 +361,27 @@ const directives = {
     ],
   },
   fontSrc: {
-    amp: [
+    ampLive: [
+      'https://gel.files.bbci.co.uk', // Reith fonts
+      'https://ws-downloads.files.bbci.co.uk', // Other WS fonts
+      'news.files.bbci.co.uk', // STY Includes
```
Comment (andrewscfc) [inline src/server/utilities/cspHeader/index.js]: shouldn't this have `https://` in, I think we want to restrict our csp to secure origins? We have the upgrade insecure flag set further down to upgrade any insecure origins used.

**Review by andrewscfc — APPROVED** [review]

**Review by chris-hinds — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
I have tested this using:

* npm run build:live: debug && npm start 
* Go to http://localhost:7080/gahuza/amakuru-50332056?renderer_env=live

I still see ths font CSP errors.Attached is the screenshot..

Firefox:
Screenshot 2020-10-26 at 13 24 20 

Chrome:

**Comment by andrewscfc** [comment]
I'm not sure we needed to make this change: https://github.com/bbc/simorgh/pull/8199/commits/6fd6ee15a87665afa4f3cf7af049e352f9a5d059

I've checked out the branch and if you look in the network tab, you see several font requests 404:
image

And if you check these against a live url they also 404: http://www.bbc.co.uk/static/media/BBCReithSans_W_Rg.woff

I _think_ the origins you have added to csp are the locations the IDT2 team have presumably changed their includes to use in a new release of IDT2, is that what @Tarqwyn said in your conversation @rhenshaw56 ?

You may need to test this bug against a page that has a newly republished IDT2 include that had this problem, @lizcameron and @jimjohnsonrollings helped me with a similar issue last week.

I hope this makes sense @paruchurisilpa and @rhenshaw56 , happy to have a call if anything need clarifying?

**Comment by rhenshaw56** [comment]
> I'm not sure we needed to make this change: [6fd6ee1](https://github.com/bbc/simorgh/commit/6fd6ee15a87665afa4f3cf7af049e352f9a5d059)
> 
> I've checked out the branch and if you look in the network tab, you see several font requests 404:
> image
> 
> And if you check these against a live url they also 404: http://www.bbc.co.uk/static/media/BBCReithSans_W_Rg.woff
> 
> I _think_ the origins you have added to csp are the locations the IDT2 team have presumably changed their includes to use in a new release of IDT2, is that what @Tarqwyn said in your conversation @rhenshaw56 ?
> 
> You may need to test this bug against a page that has a newly republished IDT2 include that had this problem, @lizcameron and @jimjohnsonrollings helped me with a similar issue last week.
> 
> I hope this makes sense @paruchurisilpa and @rhenshaw56 , happy to have a call if anything need clarifying?

@andrewscfc thanks for having a look. @Tarqwyn's recommendation was adding the `news.files` ones I added earlier. The `'self'` I added was spotted during testing. I don't seem to remember exactly if that was a recurring issue when running the live build and isn't really needed, but will confirm.

**Comment by andrewscfc** [comment]
Hey @rhenshaw56 , adding `self` gets rid of the csp error but then allows it to make a network request that 404s. I think with an up-to-date idt2, it would make requests to the new origins you added but I'm not sure?

**Comment by rhenshaw56** [comment]
> Hey @rhenshaw56 , adding `self` gets rid of the csp error but then allows it to make a network request that 404s. I think with an up-to-date idt2, it would make requests to the new origins you added but I'm not sure?

@andrewscfc you're right, those links don't exist, self policy just gets rid of the error (but I wonder why those links seem to be getting requested if they don't exist, maybe one for another issue)

**Comment by rhenshaw56** [comment]
@paruchurisilpa I just ran the local live build on the latest branch and I get the font csp violations you were seeing but none include specific 

Screen Shot 2020-10-27 at 09 54 37

**Comment by andrewscfc** [comment]
> > Hey @rhenshaw56 , adding `self` gets rid of the csp error but then allows it to make a network request that 404s. I think with an up-to-date idt2, it would make requests to the new origins you added but I'm not sure?
> 
> @andrewscfc you're right, those links don't exist, self policy just gets rid of the error (but I wonder why those links seem to be getting requested if they don't exist, maybe one for another issue)

I think the previous release of IDT2 includes make requests for these fonts, but the latest release doesn't. If we republish the IDT2 include for a particular test asset, we should expect the errors to go away.

**Comment by paruchurisilpa** [comment]
You may need to test this bug against a page that has a newly republished IDT2 include that had this problem.
 @lizcameron and @jimjohnsonrollings helped me with a similar issue last week.

``` @lizcameron or @jimjohnsonrollings would you be able to help me with this please?```

**Comment by lizcameron** [comment]
I've just republished all the IDT2 assets on this test STY https://www.test.bbc.com/gahuza/23313911

**Comment by paruchurisilpa** [comment]
* npm run build:test:debug && npm start we get the live CSP instead of that from the test environment which is a bug with our env logic which is making it use the live csp for the test environment which is why we get those violations, that probably needs to be fixed on another PR but on live it works.
* After running the local build for test ie. npm run build && debug && npm start you need to use the modHeader extension to set this value bbc-origin: https://test.bbc.com
* I dont see CSP errors anymore .

LGTM..

**Pull request merged**